core/embed/bootloader/version_check.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+
+#include "version_check.h"
+
+#include "error_handling.h"
+#include "model_version.h"
+#include "monoctr.h"
+
+// protection against bootloader downgrade
+void check_bootloader_version(void) {
```

Comment:
How about renaming this function to `ensure_bootloader_version()`, `ensure_and_check_bootloader_version()` or similar ? The verb `check` implies no side effects, like writing the bootloader version to OTP/flash memory, which confuses me a bit.

Reply:
renamed as part of https://github.com/trezor/trezor-firmware/commit/905b1005146edfe3d8d42b0dfee80c032813981f